tests/framework/test_values.py

```diff
@@ -86,3 +87,35 @@ def test_rescale_postprocessor(manager):
     odds = value.iloc[lambda x: x.index % 2 == 1]
     assert np.all(evens == from_yearly(0.5, pd.Timedelta(days=3)))
     assert np.all(odds == from_yearly(0.5, pd.Timedelta(days=5)))
+
+
+def test_step_size_post_processor(manager):
+    index = pd.Index(range(10))
+
+    pipeline = manager.register_value_producer(
+        "test",
+        source=lambda idx: [pd.Series(pd.Timedelta(days=2), index=idx)],
+        preferred_combiner=list_combiner,
+        preferred_post_processor=step_size_post_processor,
+    )
+    manager.register_value_modifier(
+        "test",
+        modifier=lambda idx: pd.Series(
+            [pd.Timedelta(days=7) if i % 2 == 0 else pd.Timedelta(days=5) for i in idx],
+            index=idx,
+        ),
+    )
+    manager.register_value_modifier(
```

Comment:
Might be worth adding a comment explicitly saying what we're testing with this modifier.